File: 11.container-with-most-water.cpp
/*
 * @lc app=leetcode id=11 lang=cpp
 *
 * [11] Container With Most Water
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int size = 0;
        while (left<right)
        {
            int h = min(height[left],height[right]);
            int w = right - left;
            size = max(size,h*w); 
            if (height[left]<height[right])
            {
                left++;
            }
            else{
                right--;
            }
            
        }
        return size;
        
        
    }
};
// @lc code=end

File: 11.container-with-most-water.py
#
# @lc app=leetcode id=11 lang=python3
#
# [11] Container With Most Water
#

# @lc code=start
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        size = (right-left)*min(height[left],height[right])
        while left<right:
            if height[left]<=height[right]:
                left+=1
                current_height = min(height[left],height[right])
                size = max(size,(right-left)*current_height)
            else:
                right-=1
                current_height = min(height[left],height[right])
                size = max(size,(right-left)*current_height)
        return size

# @lc code=end

